src/lib/bcmp/bm/bristlemouth_client.cpp

```diff
@@ -0,0 +1,71 @@
+#include "bristlemouth_client.h"
+
+// Includes for FreeRTOS
+#include "FreeRTOS.h"
+#include "configuration.h"
+#include "device_info.h"
+#include "dfu.h"
+#include "messages.h"
+#include "queue.h"
+#include "task.h"
+
+#include "bcmp_cli.h"
+#include "bm_config.h"
+#include "bm_ports.h"
+#include "bsp.h"
+#include "task_priorities.h"
+
+extern "C" {
+#include "bm_ip.h"
+#include "bristlemouth.h"
+#include "device.h"
+}
+
+#ifdef STRESS_TEST_ENABLE
+#include "stress.h"
+#endif
+
+static void adin_power_callback(bool on) { IOWrite(&ADIN_PWR, on); }
+
+void bcl_init(void) {
+  config_init();
+  //TODO: This should all be moved to user code
+  uint8_t major, minor, patch;
+  getFWVersion(&major, &minor, &patch);
+  DeviceCfg device = {
+      .node_id = getNodeId(),
+      .git_sha = getGitSHA(),
+      .device_name = getUIDStr(),
+      .version_string = getFWVersionStr(),
+      .vendor_id = 0,
+      .product_id = 0,
+      .hw_ver = 0,
+      .ver_major = major,
+      .ver_minor = minor,
+      .ver_patch = patch,
+      .sn = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'},
+  };
+
+  printf("Starting up BCL\n");
+  device_init(device);
+
+  BmErr err = bristlemouth_init(adin_power_callback);
+  if (err == BmOK) {
```

Comment:
LOOK AT THAT!!! 🕺  This is the entire code path between bm_protocol and bm_core!!!!!!